.github/workflows/cmake.yml

```diff
@@ -0,0 +1,134 @@
+name: CMake
+
+on: [push]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    name: ${{ matrix.name }}
+    # The CMake configure and build commands are platform agnostic and should work equally
+    # well on Windows or Mac.  You can convert this to a matrix build if you need
+    # cross-platform coverage.
+    # See: https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#configuring-a-build-matrix
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [
+          ubuntu-latest-gcc-10-debug,
+          ubuntu-latest-clang-10-debug,
+          ubuntu-latest-gcc-10-release,
+          ubuntu-latest-clang-10-release,
+          ubuntu-latest-clang-10-debug-libc++,
+          ubuntu-latest-clang-10-release-libc++,
+          windows-2019-cl-debug,
+          windows-2019-cl-release
+        ]
+
+        include:
+          - name: ubuntu-latest-gcc-10-debug
+            os: ubuntu-latest
+            compiler: g++-10
+            flags: ""
+            build_type: Debug
+            install_compiler: sudo apt install g++-10
+
+          - name: ubuntu-latest-clang-10-debug
+            os: ubuntu-latest
+            compiler: clang++-10
+            flags: ""
+            build_type: Debug
+            install_compiler: sudo apt install g++-10 && sudo apt install clang-10
+
+          - name: ubuntu-latest-gcc-10-release
+            os: ubuntu-latest
+            compiler: g++-10
+            flags: ""
+            build_type: Release
+            install_compiler: sudo apt install g++-10
+
+          - name: ubuntu-latest-clang-10-release
+            os: ubuntu-latest
+            compiler: clang++-10
+            flags: ""
+            build_type: Release
+            install_compiler: sudo apt install g++-10 && sudo apt install clang-10
```

Comment:
Can't you use matrix variables here?

Reply:
That is a matrix variable. The reason it's kind of complicated is b/c the installation instructions vary a bit between compilers and operating systems.